backend/cmd/headlamp.go

```diff
@@ -355,7 +356,7 @@ func createHeadlampHandler(config *HeadlampConfig) http.Handler {
 
 	plugins.PopulatePluginsCache(config.staticPluginDir, config.pluginDir, config.cache)
 
-	if !config.useInCluster {
+	if !config.useInCluster && config.watchPluginsChanges {
```

Comment:
I think this is a problem. IIRC there was someone asking for a way to watch plugins while -in-cluster, right? In that case, this would never be possible.
So we should likely have a way to detect if watch-plugins-changes has been overridden by the user directly, and if not, then it takes the !useInCluster value.
Then in this line all we need to check if the value of watchPluginsChanges.

Reply:
It should be `!config.useInCluster || config.watchPluginsChanges`  I think.

I believe the request in the issue was using the container, but running it out of cluster?